### ISSUE TITLE
Add support for RFC 7405 %s and %i prefixes for strings

### ISSUE THREAD

**Issue by mxmlnkn** [michaelrsweet/abnf#5]
Trying out the [date time specification](https://tools.ietf.org/html/rfc3339#section-5.6), I noticed that the `T` separator gets replaced with a lower capital `t` for the `case-sensitive` mode. This makes no sense to me. So, a smaller example would be:

```ABNF
date-fullyear   = 4DIGIT
time-hour       = 2DIGIT
date-time       = date-fullyear "T" time-hour
```

Given to `https://abnf.msweet.org/`, returns `date-time: ^[0-9]{4,4}t[0-9]{2,2}$` for `date-time` rule with RGEX mode `case-sensitive`.
However, it should be: `date-time: ^[0-9]{4,4}T[0-9]{2,2}$`

**Comment by michaelrsweet** [comment]
Might be an issue with the way I'm loading the character strings - ordinarily strings like that are case-insensitive so it is "safe" to downcase. But in my defense, that ABNF notes that lowercase "T" is allowed in time values, too... :)

**Comment by mxmlnkn** [comment]
You are right. After writing the issue I noticed that the RFC explicitly restates the ABNF rules that "T" is case insensitive. So, it might be an issue when you read the input as case insensitive but then offer an option for case sensitivity.

In the first place, I found the case options very confusing. As a first time user I would've just wanted an option "do it just as the ABNF says". Only after looking into it, I knew that it corresponds to case insensitive as there is no case sensitive string in the [original ABNF](https://tools.ietf.org/html/rfc5234) instead they would have to be specified as ASCII numbers which currently also does not seem to work as I get `bad character value` for `h = %d101`. 

There even is a [variation](https://tools.ietf.org/html/rfc7405) which allows specifying case sensitivity for string. But while `h = "Ab"` works, I also get `^ Bad character value.` for `h = %s"Ab"`. Not sure whether the percent sign is buggy or whether you are testing changes directly on the public web site?

**Comment by michaelrsweet** [comment]
WRT `%s`, that and `%i` are an extension to RFC 5234 defined in RFC 7405, which this code currently does not support. Retitling this bug for future implementation...

**Comment by gibson042** [comment]
Another request for this functionality: https://github.com/unicode-org/message-format-wg/issues/501#issuecomment-1794517256
